unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/LoadingFeedbackController.cs

```diff
@@ -35,20 +37,33 @@ private class SceneLoadingStatus
 
         public LoadingFeedbackController()
         {
-            loadedScenes = new List<SceneLoadingStatus>();
+            DataStore.i.featureFlags.flags.OnChange += FeatureFlagsSet;
+        }
+
+        private void FeatureFlagsSet(FeatureFlag current, FeatureFlag previous)
+        {
+            isDecoupledLoadingScreenEnabled = current.IsFeatureEnabled(DataStore.i.featureFlags.DECOUPLED_LOADING_SCREEN_FF);
+            if (!isDecoupledLoadingScreenEnabled)
+            {
+                loadedScenes = new List<SceneLoadingStatus>();
 
-            Environment.i.world.sceneController.OnNewSceneAdded += SceneController_OnNewSceneAdded;
-            GLTFComponent.OnDownloadingProgressUpdate += GLTFComponent_OnDownloadingProgressUpdate;
-            AssetPromise_AB.OnDownloadingProgressUpdate += AssetPromise_AB_OnDownloadingProgressUpdate;
-            CommonScriptableObjects.rendererState.OnChange += RendererState_OnChange;
+                Environment.i.world.sceneController.OnNewSceneAdded += SceneController_OnNewSceneAdded;
+                GLTFComponent.OnDownloadingProgressUpdate += GLTFComponent_OnDownloadingProgressUpdate;
+                AssetPromise_AB.OnDownloadingProgressUpdate += AssetPromise_AB_OnDownloadingProgressUpdate;
+                CommonScriptableObjects.rendererState.OnChange += RendererState_OnChange;
+            }
+            DataStore.i.featureFlags.flags.OnChange -= FeatureFlagsSet;
```

Comment:
I would recommend to put this unsubscribtion at the start of this method

Reply:
Done